runtime/int.go

```diff
@@ -98,6 +98,22 @@ func intEq(f *Frame, v, w *Object) (*Object, *BaseException) {
 	return intCompare(compareOpEq, toIntUnsafe(v), w), nil
 }
 
+func intFloorDiv(f *Frame, v, w *Object) (*Object, *BaseException) {
+	if !v.isInstance(IntType) {
+		return nil, f.RaiseType(TypeErrorType, fmt.Sprintf("descriptor '__floordiv__' requires a 'int' object but received a '%s'", v.typ.Name()))
```

Comment:
The slot delegation code guarantees that the first argument to slot functions is of the correct type, so no need to check here.